### PR TITLE
Add tests for possible float64 transmogrification transgressions

### PR DESCRIPTION
**Pull request by mpalmer**
Make sure that infinities, subnormals, and -0 all convert correctly.
Found a bug in how -0 is handled, and fixed it.

Quickcheck tests are now failing, but I'm not smart enough to fix problems with anything derived from Haskell.

### CODE REVIEW

File: src/encode.rs [hunk 1]
```diff
@@ -22,6 +22,9 @@ impl From<f32> for OrePlaintext<u64> {
 
 impl From<f64> for OrePlaintext<u64> {
     fn from(term: f64) -> OrePlaintext<u64> {
+        if term == 0f64 || term == -0f64 {
+            return OrePlaintext(0u64);
+        }
```
Comment (freshtonic) [inline src/encode.rs]: I *think* this means we will end up with an unstable sort order for records indexed on a float when we do "order by" in queries in the presence of both 0 and -0. Essentially, the order that records come back will be non-deterministic.

Comment (freshtonic) [reply src/encode.rs]: When 0 and -0 are massaged into a u64, what were the two u64 representations? I'm guessing that they were not adjacent and hence the weird comparison result.

**Comment by freshtonic** [comment]
The `encode::tests::test_infinity_sorts_correctly` test is failing for me too.

File: src/encode.rs (hunk 1 above)
Comment (freshtonic) [reply src/encode.rs]: Actually, I'm ok with this. It looks like I was already going down this path somewhat but didn't actually fix the encoding.

**Comment by freshtonic** [comment]
> The `encode::tests::test_infinity_sorts_correctly` test is failing for me too.

Tested under ARM64 and x86_64 (under Rosetta - which may be imperfect).

The line that failed is `src/encode.rs:114`

**Comment by freshtonic** [comment]
This change will fix the failing quickcheck test:

```
diff --git a/src/encode.rs b/src/encode.rs
index 69420f1..28aac77 100644
--- a/src/encode.rs
+++ b/src/encode.rs
@@ -138,7 +138,7 @@ mod tests {

     quickcheck! {
         fn roundtrip_one_f64(x: f64) -> TestResult {
-            if !x.is_nan() && x.is_finite() {
+            if !x.is_nan() && x != -0.0 {
                 TestResult::from_bool(x == decode_u64_to_f64(x.into()))
             } else {
                 TestResult::discard()
```

**Comment by mpalmer** [comment]
Thanks for that quickcheck fix, did the trick nicely.  Turns out my zero fix had broken infinity, but that's sorted now, too.

**Review by freshtonic — APPROVED** [review]
Thought I'd already approved this, derp derp.

**Pull request merged**